src/main/java/com/application/views/masterdetail/SkigebietForm.java

```diff
@@ -0,0 +1,170 @@
+    TextField URLTicketpage = new TextField("Ticketpreise");
+    IntegerField avalancheWarningLevel = new IntegerField("aktuelle Warnsufe");
+    TextField URLImageFront = new TextField("Titelbild");
+    TextField URLImageSlope = new TextField("Pistenbild");
+
+    Button save = new Button("Save");
+    Button delete = new Button("Delete");
+    Button close = new Button("Cancel");
+    private SkiResort skiResort;
+
+
+    public SkigebietForm() {
+        addClassName("Skigebiet-Form");
+        binder.bindInstanceFields(this);
+
+        add(
+                name,
+                region,
+                operator,
+                address,
+                zip,
+                city,
+                heightMin,
+                heightMax,
+                totalLenght,
+                ropeways,
+                pos_lon,
+                pos_lat,
+                dateSeasonStart,
+                dateSeasonEnd,
+                timeServiceStart,
+                timeServiceEnd,
+                currentUtilizationPercent,
+                userRating,
+                weatherCurrentWindspeed,
+                weatherCurrentTemperature,
+                weatherCurrentSnowfallForecastPercent,
+                weatherCurrentSnowfallForecastAmountMM,
+                weatherDatetimeLastRead,
+                snowDepthMin,
+                snowDepthMax,
+                amountFreshSnow,
+                dateLastSnowfall,
+                URLTicketpage,
+                avalancheWarningLevel,
+                URLImageFront,
+                URLImageSlope,
+                createButtonsLayout()
+        );
+    }
+
+    public void setSkiResort(SkiResort skiResort){
+        this.skiResort = skiResort;
+        binder.readBean(skiResort);
+    }
+
+
+    private HorizontalLayout  createButtonsLayout(){
+        save.addThemeVariants(ButtonVariant.LUMO_PRIMARY);
+        delete.addThemeVariants(ButtonVariant.LUMO_ERROR);
+        close.addThemeVariants(ButtonVariant.LUMO_TERTIARY);
+
+
+        save.addClickListener(event -> validateAndSave());
+        delete.addClickListener(event -> fireEvent(new DeleteEvent(this, skiResort)));
+        close.addClickListener(event -> fireEvent(new CloseEvent(this)));
+
+        save.addClickShortcut(Key.ENTER);
+        close.addClickShortcut(Key.ESCAPE);
+
+        return new HorizontalLayout(save, delete, close);
+    }
+
+    private void validateAndSave(){
+        try{
+            binder.writeBean(skiResort);
+            fireEvent(new SaveEvent(this, skiResort));
+        } catch (ValidationException e){
+            e.printStackTrace();
+        }
+    }
+
+    public static abstract class SkigebietFormEvent extends ComponentEvent<SkigebietForm>{
+        private SkiResort skiResort;
+        protected SkigebietFormEvent(SkigebietForm source, SkiResort contact){
+            super(source, false);
+            this.skiResort = contact;
+        }
+        public SkiResort getContact(){
```

Comment:
getContact in getSkiResort umbenennen, macht hier mehr Sinn

Reply:
erledigt